shtdlib.sh

```diff
@@ -287,6 +296,37 @@ function shopt_decorator {
     exit 127
 }
 
+# Test decorator
+# Forces a function to be executed in all bash variants using the bashtester
+# submodule and containers. Requires docker to be installed and git submodules
+# to be present and up do date.
+# To use this add a line like the following (without #) as the first line of a function
+# test_decorator "${FUNCNAME[0]}" "${@:-}" && return
+# test_decorator "${FUNCNAME[0]}" "${@:-}" && return || conditional_exit_on_fail 121 "Failed to run ${FUNCNAME[0]} with test_decorator"
+
+# To specify a different set of bash versions set supported-bash_versions to a
+# space separated string of the supported versions.
+function test_decorator {
+    # If not running in a container
+    if [ "${FUNCNAME[0]}" != "${FUNCNAME[2]}" ] && ! cat /proc/1/cgroup | grep -q docker; then
```

Comment:
Should we allow testing to occur without being in a container? Might help someone test locally if they don't have a docker ENV

Reply:
That decorator is used specifically to run stuff in containers but we could add something to the test_shtdlib function to pick if we want it run locally or in a container.

Reply:
Now accepts local as a parameter to test_shtdlib and the version :)